### PR TITLE
Fix Equal schema to take integral input types

### PR DESCRIPTION
**Pull request by bddppq**
@tianleiwu For test update

@ezyang I'm actually not sure whether this should fall into breaking change or schema fix

### CODE REVIEW

File: onnx/backend/test/case/node/equal.py [hunk 1]
```diff
@@ -20,8 +20,8 @@ def export():
             outputs=['less'],
```
Comment (tianleiwu) [inline onnx/backend/test/case/node/equal.py]: It is better to change the output node name from 'less' to 'z' or 'equal'.

**Comment by tianleiwu** [comment]
Good catch of schema issue of Equal.

For Greater and Less, I think we can add support for integral types as well.

File: onnx/defs/logical/defs.cc [hunk 2]
```diff
@@ -66,8 +66,8 @@ OPERATOR_SCHEMA(Less)
 
 OPERATOR_SCHEMA(Equal)
     .FillUsing(BinaryLogicDocGenerator("equal"))
-    .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
-                    "Constrains input to float tensors.")
+    .TypeConstraint("T", { "tensor(bool)", "tensor(int32)", "tensor(int64)" },
+                    "Constrains input to integral tensors.")
```
Comment (ezyang) [inline onnx/defs/logical/defs.cc]: I'm confused, doesn't this diff remove support for equality test on floating point?

Comment (bddppq) [reply onnx/defs/logical/defs.cc]: Yeah I think it probably doesn't make too much sense to provide equality test on floats

**Pull request merged**